jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+will fail.
+
+3. Validation that no extra fields are present in the file. Comments are allowed, but any unexpected 
+lines will cause validation to fail.
+
+If validation of `Autorelease.yaml` fails for any reason, the build is not performed nothing gets deployed, 
+and GitHub is notified of the failure.
+
+==== Deployment of a successful release
+
+When Karl merges a commit into the `master` branch, that merge commit is commit number 150, and has the 
+SHA `1a2b3c4`. The following takes place:
+
+* A webhook is sent to `${trusted-ci-server}`, and `Autorelease.yaml` is validated.
+* Once validation passes, a build is performed. If the build passes all its tests, a release 
+is generated. In our example, that release number would be `3.1.4-150-1a2b3c4`
+* The built plugin gets deployed to Artifactory
+* The resulting plugin appears on the Jenkins Update Center, as Karl Plugin version `3.1.4-150-1a2b3c4`.
+
+== Motivation
+
+At present, Jenkins plugins are, typically, not released on a continous basis. They 
+are also not released from a single source of truth, such as a trusted Continuous 
+Integration server like link:https://ci.jenkins.io[https://ci.jenkins.io]. 
+
+The notion of continuous delivery of plugin releases has been discussed previously <<footnote-1,^(1)^>>.
+Considering that Jenkins is a system used to facilitate Continuous Delivery for many users, it makes 
+sense -- and builds credibility -- for the Jenkins developer community to adopt this same practice. 
+
+Having a centralized release system made available to plugin maintainers also provides additional 
+confidence that security best practices are being followed. 
+
+Continuous delivery from trusted CI is something which plugin maintainers can opt into, but is 
+not required. If a plugin maintainer chooses to continue to follow their own path for releasing 
+versions of their code, they remain free to do so.
+
+== Reasoning
+
+=== What's with the version numbering?
+
+Jenkins plugin maintainers are already familiar with the way that Incrementals appends a commit 
+number, plus a SHA, to version numbers. These mechanically-generated version numbers offer 
+the ability to predictably sort them, so that external systems, such as the Jenkins update center, 
+can correctly publish the "newest" version. The addition of a merge SHA also allows for at-a-glance 
+feedback to tell people which commits went into the release.
+
+WARNING: We know the problem of consistent version numbering, and the resulting ability to 
+sort versions, is important. The scheme discussed here is only one of several possible 
+choices. We expect some _lively debate_ around this topic. But it's very important to ensure 
+that versions can be easily sorted by systems such as the Jenkins update center. Also under 
+consideration is defining a fixed number of fields in version numbers, and considering any 
+deviation from that a validation failure.
+
+=== Why YAML?
```

Comment:
Can probably be dropped. I cannot imagine anyone raising an eyebrow over _this_ aspect.

Reply:
Yeah, maybe you're right.

Reply:
In case some asks why not JSON or Groovy?